src/commands/code-analyzer/run.ts

```diff
@@ -1,19 +1,74 @@
-import {SfCommand} from '@salesforce/sf-plugins-core';
+import {Flags, SfCommand} from '@salesforce/sf-plugins-core';
+import {RunAction} from '../../lib/actions/RunAction';
+import {BundleName, getMessage} from '../../lib/messages';
 
-export default class Run extends SfCommand<void> {
+export default class RunCommand extends SfCommand<void> {
 	// We don't need the `--json` output for this command.
 	public static readonly enableJsonFlag = false;
-	public static readonly summary = 'summary boop'; // TODO
-	public static readonly description = 'description boop'; // TODO
-	public static readonly examples = [ // TODO
-		'example1 boop',
-		'example2 boop'
+	public static readonly summary = getMessage(BundleName.RunCommand, 'command.summary');
+	public static readonly description = getMessage(BundleName.RunCommand, 'command.description');
+	public static readonly examples = [
+		getMessage(BundleName.RunCommand, 'command.examples')
 	];
 
-	public static readonly flags = {};
+	public static readonly flags = {
+		// === Flags pertaining to targeting ===
+		workspace: Flags.string({
+			summary: getMessage(BundleName.RunCommand, 'flags.workspace.summary'),
+			char: 'w',
+			multiple: true,
+			delimiter: ',',
+			default: ['.']
+		}),
+		'path-start': Flags.string({
+			summary: getMessage(BundleName.RunCommand, 'flags.path-start.summary'),
+			char: 's',
+			multiple: true,
+			delimiter: ','
+		}),
+		// === Flags pertaining to rule selection ===
+		'rule-selector': Flags.string({
+			summary: getMessage(BundleName.RunCommand, 'flags.rule-selector.summary'),
+			char: 'r',
+			multiple: true,
+			delimiter: ',',
+			default: ["recommended"]
+		}),
+		// === Flags pertaining to output ===
+		'severity-threshold': Flags.string({
+			summary: getMessage(BundleName.RunCommand, 'flags.severity-threshold.summary'),
+			char: 't',
+			options: [
+				'1', 'critical',
+				'2', 'high',
+				'3', 'moderate',
+				'4', 'low',
+				'5', 'info'
+			]
+		}),
+		view: Flags.string({
+			summary: getMessage(BundleName.RunCommand, 'flags.view.summary'),
+			char: 'v',
+			options: ['table', 'detail'] // TODO: Should probably be enum?
+		}),
+		'output-file': Flags.string({
```

Comment:
There's no easy way to enforce patterns at the flag-declaration level, so enforcing that the file has the desired extension will be done either in a new method at this layer, or in the Action layer. Probably the latter, for testability.

Reply:
Yes this is correct. It's easy enough to do validation later on things like (checking if the parent folder exists so that we can actually write the file). Basically same stuff we do today.